includes/class-wc-stripe-webhook-handler.php

```diff
@@ -525,15 +525,33 @@ public function process_webhook_capture( $notification ) {
 	 * @param object $notification
 	 */
 	public function process_webhook_charge_succeeded( $notification ) {
-		// The following payment methods are synchronous so does not need to be handle via webhook.
-		if ( ( isset( $notification->data->object->source->type ) && WC_Stripe_Payment_Methods::CARD === $notification->data->object->source->type ) || ( isset( $notification->data->object->source->type ) && 'three_d_secure' === $notification->data->object->source->type ) ) {
+		if ( empty( $notification->data->object ) ) {
+			WC_Stripe_Logger::log( 'Missing charge object in charge.succeeded webhook' );
```

Comment:
(nit) I wonder if including the notification id would be helpful here.


```suggestion
			'Missing charge object in charge.succeeded webhook, Event ID: %s', $notification->id ?? 'unknown'```

Reply:
Good idea!
```suggestion
			WC_Stripe_Logger::log( 'Missing charge object in charge.succeeded webhook, Event ID: %s', $notification->id ?? 'unknown' );
```